trino/serial.go

```diff
@@ -35,6 +35,55 @@ func (e UnsupportedArgError) Error() string {
 // If another string format is used it will error to serialise
 type Numeric string
 
+// Date indicates we want a Date type in Trino from a Golang time.
+type trinoDate struct {
+	year  int
+	month time.Month
+	day   int
+}
+
+func Date(year int, month time.Month, day int) trinoDate {
```

Comment:
All exported functions should have comments, maybe something like:
```suggestion
// Date creates a representation of a Trino Date type.
func Date(year int, month time.Month, day int) trinoDate {
```